### PR TITLE
add ConnectionTTL to plugin-cfg.xml, along with persistTimeoutReduction parm in pluginConfiguration

### LINKED ISSUE #8697: add ability to specify ConnectionTTL for transport element of plugin
With this change, ConnectionTTL is added to the Transport lines of plugin-cfg.xml.  ConnectionTTL is a calculated value: persistTimeout (from the applicable httpOptions) minus persistTimeoutReduction.  persistTimeoutReduction is a new option on the pluginConfiguration statement, for example:
<pluginConfiguration persistTimeoutReduction="5s" />

The default value of persistTimeoutReduction is "2s".

### PR DESCRIPTION
**Pull request by epj**
fixes #8697

### CODE REVIEW

**Comment by epj** [comment]
With this proposed change, the reduction to the ConnectionTTL is specified in server.xml, not ConnectonTTL directly:
<pluginConfiguration PersistTimeOut_Reduction="5" />

The default value of PersistTimeOut_Reduction is 2.  When PersistTimeOut_Reduction is specified with a value other than 2 (within the range 0 to 30) then ConnectionTTL is added to the transport statements in plugin-cfg.xml with the specified reduction.

**Comment by epj** [comment]
#libbybot

**Comment by epj** [comment]
#libbybot

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype-mbeans.properties [hunk 1]
```diff
@@ -48,6 +48,9 @@ plugin.http.endpoint.desc=Specify the identifier of the http endpoint to include
 plugin.connectTimeout.name=Connection timeout
 plugin.connectTimeout.desc=Identifies the maximum amount of time that the application server should maintain a connection with the web server.
 
+plugin.persistTimeOut_Reduction.name=Reduce PersistTimeOut for ConnectionTTL
+plugin.persistTimeOut_Reduction.desc=Identifies the amount of time to subtract from the PersistTimeOut value for ConnectionTTL.  
+
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype-mbeans.properties]: `Identifies the amount of time to subtract from the PersistTimeOut value for ConnectionTTL.`
suggested edit:

`Specifies the amount of time to subtract from the value of the persistTimeOut parameter to calculate the value for the ConnectionTTL parameter.`

**Comment by epj** [comment]
#libbybot

**Comment by epj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Comment by epj** [comment]
#libbybot

**Comment by epj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by epj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.1,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by epj** [comment]
#libbybot

**Pull request merged**